components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/UserResource.java

```diff
@@ -348,13 +352,16 @@ public Response patchUser(@PathParam(SCIMConstants.CommonSchemaConstants.ID) Str
             // obtain the user store manager
             UserManager userManager = IdentitySCIMManager.getInstance().getUserManager();
 
-            String superAdminID = AdminAttributeUtil.getSuperAdminID();
-            String loggedInUser = SCIMCommonUtils.getLoggedInUserID();
-            if ((superAdminID.equals(id)) && (!loggedInUser.equals(id))) {
-                if (LOG.isDebugEnabled()) {
-                    LOG.debug("Do not have permission to patch SuperAdmin user.");
+            // Skipping this validation if the request comes from a sub organization.
+            if (!SCIMCommonUtils.isOrganization(getTenantDomain())) {
+                String superAdminID = AdminAttributeUtil.getSuperAdminID();
+                String loggedInUser = SCIMCommonUtils.getLoggedInUserID();
+                if ((superAdminID.equals(id)) && (!loggedInUser.equals(id))) {
+                    if (LOG.isDebugEnabled()) {
```

Comment:
We can remove this if check and only have Log.debug line since there are no string concats